docs/current_docs/quickstart/basics/basics.mdx

```diff
@@ -0,0 +1,436 @@
+
+Here's a revision of the previous example which splits it into two smaller functions: one to build the container, and one to publish it. The builder function accepts the container image string as an argument and returns a `Container` object. The publisher function accepts a `Container` as argument and returns the image identifier as a string.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go
+type Basics struct{}
+
+func (m *Basics) Build(
+	// +default "alpine:latest"
+	image string,
+) *dagger.Container {
+	return dag.Container().
+		From(image).
+		WithNewFile("/hi.txt", "Hello from Dagger!")
+}
+
+func (m *Basics) Publish(
+	ctx context.Context,
+	// +default "alpine:latest"
+	image string,
+) (string, error) {
+	return m.FooBuild(image).
+		WithEntrypoint([]string{"cat", "/hi.txt"}).
+		Publish(ctx, "ttl.sh/hello")
+}
+```
+
+</TabItem>
+<TabItem value="Python">
+
+FIXME
+
+</TabItem>
+<TabItem value="TypeScript">
+
+FIXME
+
+</TabItem>
+<TabItem value="PHP">
+
+FIXME
+
+</TabItem>
+<TabItem value="Java">
+
+FIXME
+
+</TabItem>
+</Tabs>
+
+[FIXME recording]
+
+:::tip FUNCTION NAMES
+When calling Dagger Functions, all names (functions, arguments, fields, etc.) are converted into a shell-friendly "kebab-case" style. This is why a Dagger Function named `FooPublish` in Go, `foo_publish` in Python and `fooPublish` in TypeScript/PHP/Java is called as `foo-publish` in Dagger Shell or on the command-line.
+:::
+
+## Speeding things up with caching
+
+One of Dagger's most powerful features is its ability to cache data across workflow runs. Dagger caches two types of data:
+
+- Layers: Build instructions and the results of some API calls.
+- Volumes: Contents of Dagger filesystem volumes.
+
+Taken together, these two types of caching can significantly reduce execution times. Here's an example: a Dagger Function that creates a cache volume to store the packages installed by `apt`:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go
+type Basics struct{}
+
+func (m *Basics) Foo(ctx context.Context) (*dagger.Container, error) {
+	aptCache := dag.CacheVolume("apt-cache")
+	return dag.Container().
+		From("debian:latest").
+		WithMountedCache("/var/cache/apt/archives", aptCache).
+		WithExec([]string{"apt-get", "update"}).
+		WithExec([]string{"apt-get", "install", "--yes", "maven", "mariadb-server"}).
+		Sync(ctx)
+}
+```
+
+</TabItem>
+<TabItem value="Python">
+
+FIXME
+
+</TabItem>
+<TabItem value="TypeScript">
+
+FIXME
+
+</TabItem>
+<TabItem value="PHP">
+
+FIXME
+
+</TabItem>
+<TabItem value="Java">
+
+FIXME
+
+</TabItem>
+</Tabs>
+
+Notice that when you call this Dagger Function multiple times, the second and subsequent runs are drastically faster than the first, since Dagger automatically reuses cached instructions and files from the cache.
+
+[FIXME recording]
+
+## Using the Daggerverse
```

Comment:
What if we inverse this to be something like "installing other modules" or something, show how a function from the previous section gets added into your `dag` client, and mention daggerverse in the description? And then also bump it above caching

Reply:
I didn't understand this suggestion. I'm not really sure how to show the previous function getting added into the client. Can you clarify?

I've taken a different approach, adding two modules and showing them in the client. Does this work?

Also moved it above the caching section.

Reply:
Yeah that's fine for now I think! What I was suggesting would have been something like

The previous snippet showed a module `Basics` with a function `Build()`. If we create a new module and install `Basics` as a dependency, we can now call it's functions with `ctr := dag.Basics().Build()` to show that dependencies aren't a special thing related to Daggerverse, it's any Dagger module

Reply:
Ok I see what you mean. The reason I did it this way is because it lets me introduce the Daggerverse early